ui/pages/confirmations/hooks/alerts/transactions/useInsufficientBalanceAlerts.ts

```diff
@@ -48,12 +49,10 @@ export function useInsufficientBalanceAlerts(): Alert[] {
     selectTransactionFeeById(state, transactionId),
   );
 
-  const [multichainNetworks, evmNetworks] = useSelector(
-    getMultichainNetworkConfigurationsByChainId,
+  const networkConfiguration: Record<Hex, NetworkConfiguration> = useSelector((state) =>
+    selectNetworkConfigurationByChainId(state, chainId),
   );
-  const nativeCurrency = (
-    multichainNetworks[chainId as CaipChainId] ?? evmNetworks[chainId]
-  )?.nativeCurrency;
+  const nativeCurrency = networkConfiguration?.nativeCurrency;
```

Comment:
Above is fix for feedback here: https://github.com/MetaMask/metamask-extension/pull/32673#discussion_r2081333587

Reply:
Do we need it to be this PR? I am asking as we are aiming to cherry pick this PR into 12.17.3... And the above was introduced in a PR that was cp into 12.18.

Reply:
I am not sure if we are going to cherry-pick this, but let me move above change to separate PR anyways.

Reply:
PR is updated